src/AppInstallerCommonCore/ThreadGlobals.cpp

```diff
@@ -0,0 +1,73 @@
+#include "pch.h"
+#include "Public/winget/ThreadGlobals.h"
+
+namespace AppInstaller::ThreadLocalStorage
+{
+    using namespace AppInstaller::Logging;
+
+    DiagnosticLogger& ThreadGlobals::GetDiagnosticLogger()
+    {
+        return *(m_pDiagnosticLogger);
+    }
+
+    TelemetryTraceLogger& ThreadGlobals::GetTelemetryLogger()
+    {
+        return *(m_pTelemetryLogger);
+    }
+
+    PreviousThreadGlobals* ThreadGlobals::SetForCurrentThread()
+    {
+        Initialize();
+
+        std::unique_ptr<PreviousThreadGlobals> p_prevThreadGlobals = std::make_unique<PreviousThreadGlobals>(SetOrGetThreadGlobals(true, this));
+
+        return p_prevThreadGlobals.release();
+    }
+
+    void ThreadGlobals::Initialize()
+    {
+        try
+        {
+            std::call_once(loggerInitOnceFlag, [this]()
+                {
+                    m_pDiagnosticLogger = std::make_unique<DiagnosticLogger>();
+                    m_pTelemetryLogger = std::make_unique<TelemetryTraceLogger>();
+
+                    if (m_pTelemetryLogger)
```

Comment:
nit: `make_unique` will throw or there will be an object; this if statement is always going to be true.